server/proxy/uninstall.go

```diff
@@ -41,6 +41,17 @@ func (p *Proxy) UninstallApp(r *incoming.Request, cc apps.Context, appID apps.Ap
 		}
 	}
 
+	// If the app's cleanup fails in the middle, disable the app and return the error.
+	defer func() {
+		if err == nil {
+			return
+		}
+		r.Log.WithError(err).Errorf("Failed to uninstall app %s: %s", appID, err)
```

Comment:
Nitpick: Why does the error need to be included two times? In  `WithError(err)` and via `Errorf`?

Reply:
to make JSON-less text readable (i.e. include the essentials). However, perhaps errors.Cause(err).Error() is a better one to include, leaving the full one in JSON?